### PR TITLE
feat(backend): notifications for groups

### LINKED ISSUE #6097: 🚀 [Feature] Backend / Webapp – Group / Notification – Send User a Notification that their Rolle has been Changed
## 🚀 Feature

Send User a Notification that their Rolle has been Changed.

```gherkin
Feature: Notification on member roll change
  As a user
  I want to be notified
  if I get a none pending membership roll in a group I entered
  or if I'm added to a group.
  Probably I like to be notified on all changes of my member roll
  as well if I enter a group or be added
  or if I get thrown out of a group or leave it.
```

### PR DESCRIPTION
**Pull request by Mogge**
### Issues
- fixes #6097 

### Todo

- [X] None

### CODE REVIEW

**Review by Elweyn — APPROVED** [review]
Todos create Ticket for:
- [ ] Change notification tab height to match buttons

**Review by Tirokk — CHANGES_REQUESTED** [review]
I have found some errors @Mogge , but had no time yet to write them down.

Please contact me.

**Review by Elweyn — APPROVED** [review]

**Review by ulfgebhardt — DISMISSED** [review]

File: backend/src/schema/resolvers/notifications.spec.js [hunk 1]
```diff
@@ -397,18 +397,20 @@ describe('given some notifications', () => {
...
-              read: true,
-            },
-            {
-              createdAt: '2019-08-31T17:33:48.651Z',
-              from: { __typename: 'Post', content: 'You have been mentioned in a post' },
-              read: true,
-            },
-          ])
+          expect(response.data.markAllAsRead).toEqual(
+            expect.arrayContaining([
```
Comment (Tirokk) [inline backend/src/schema/resolvers/notifications.spec.js:401]: So you didn't like to test this for group notifications …

That's okay.

**Review by Tirokk — APPROVED** [review]
Hey @Mogge 

very cool done dear! 😍

I approve 🚀🚀🚀💫💫

PS: Just one little suggestion.

**Pull request merged**